<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        <meta name="author" content="Yang Fang">
        
        <link rel="shortcut icon" href="../img/favicon.ico">
        <title>Usage - Physpetools</title>
        <link href="../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../css/font-awesome.min.css" rel="stylesheet">
        <link href="../css/base.css" rel="stylesheet">
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.4.2/respond.min.js"></script>
        <![endif]-->

        <script src="../js/jquery-1.10.2.min.js" defer></script>
        <script src="../js/bootstrap-3.0.3.min.js" defer></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
        <script>hljs.initHighlightingOnLoad();</script> 
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
            <div class="container">

                <!-- Collapsed navigation -->
                <div class="navbar-header">
                    <!-- Expander button -->
                    <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                        <span class="sr-only">Toggle navigation</span>
                        <span class="icon-bar"></span>
                        <span class="icon-bar"></span>
                        <span class="icon-bar"></span>
                    </button>
                    <a class="navbar-brand" href="..">Physpetools</a>
                </div>

                <!-- Expanded navigation -->
                <div class="navbar-collapse collapse">
                        <!-- Main navigation -->
                        <ul class="nav navbar-nav">
                            <li >
                                <a href="..">Home</a>
                            </li>
                            <li >
                                <a href="../install/">Install</a>
                            </li>
                            <li class="dropdown active">
                                <a href="#" class="dropdown-toggle" data-toggle="dropdown">Documents <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li class="active">
    <a href="./">Usage</a>
</li>
                                    
<li >
    <a href="../tutorial/">Tutorial</a>
</li>
                                </ul>
                            </li>
                            <li >
                                <a href="../faq/">FAQ</a>
                            </li>
                        </ul>

                    <ul class="nav navbar-nav navbar-right">
                        <li>
                            <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                                <i class="fa fa-search"></i> Search
                            </a>
                        </li>
                            <li >
                                <a rel="next" href="../install/">
                                    <i class="fa fa-arrow-left"></i> Previous
                                </a>
                            </li>
                            <li >
                                <a rel="prev" href="../tutorial/">
                                    Next <i class="fa fa-arrow-right"></i>
                                </a>
                            </li>
                            <li>
                                <a href="https://github.com/xiaofeiyangyang/physpetools/edit/master/docs/usage.md"><i class="fa fa-github"></i> Edit on GitHub</a>
                            </li>
                    </ul>
                </div>
            </div>
        </div>

        <div class="container">
                <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
        <li class="main active"><a href="#usage-and-examples">Usage and Examples</a></li>
            <li><a href="#autobuild">autobuild</a></li>
            <li><a href="#build">build</a></li>
            <li><a href="#combine">combine</a></li>
            <li><a href="#iview">iview</a></li>
            <li><a href="#check">check</a></li>
    </ul>
</div></div>
                <div class="col-md-9" role="main">

<h1 id="usage-and-examples">Usage and Examples</h1>
<h2 id="autobuild">autobuild</h2>
<p>The input of <code>autobuild</code> module is a TXT file containing <a href="http://www.genome.jp/kegg/catalog/org_list.html">KEGG</a> abbreviated species names, for example <a href="../example/organism_example_list.txt">organism_example_list</a>.</p>
<pre><code class="bash">$ PhySpeTree -i autobuild organism_example_list.txt [options]*
</code></pre>

<h3 id="options">options</h3>
<table>
<thead>
<tr>
<th align="left">option</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">-h</td>
<td align="left">Print help message and exits.</td>
</tr>
<tr>
<td align="left">-i</td>
<td align="left">Input a TXT file containing abbreviated species names.</td>
</tr>
<tr>
<td align="left">-o</td>
<td align="left">A directory to store outputs. The default is "Outdata".</td>
</tr>
<tr>
<td align="left">-t</td>
<td align="left">Number of processing threads (CPUs). The default is 1.</td>
</tr>
<tr>
<td align="left">-e</td>
<td align="left">FASTA format files to extend the tree with the --ehcp or --esrna option.</td>
</tr>
<tr>
<td align="left">-db</td>
<td align="left">The absolute path for local database.</td>
</tr>
<tr>
<td align="left">--hcp</td>
<td align="left">HCP (highly conserved protein) method (default).</td>
</tr>
<tr>
<td align="left">--ehcp</td>
<td align="left">HCP method with extended HCP sequences.</td>
</tr>
<tr>
<td align="left">--srna</td>
<td align="left">SSU method.</td>
</tr>
<tr>
<td align="left">--esrna</td>
<td align="left">SSU rRNA method with extended SSU rRNA sequences.</td>
</tr>
</tbody>
</table>
<h3 id="example">Example</h3>
<p>Download the example input file:</p>
<pre><code class="bash">$ wget &quot;https://yangfangs.github.io/physpetools/example/organism_example_list.txt&quot;

--2016-10-29 19:41:53--  https://yangfangs.github.io/physpetools/example/organism_example_list.txt
Resolving yangfangs.github.io (yangfangs.github.io)... 151.101.24.133
Connecting to yangfangs.github.io (yangfangs.github.io)|151.101.24.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 39 [text/plain]
Saving to: ‘organism_example_list.txt’

organism_example_list.txt     100%[==============================================&gt;]      39  --.-KB/s    in 0s      

2016-10-29 19:41:54 (19.0 MB/s) - ‘organism_example_list.txt’ saved [39/39]

$ cat organism_example_list.txt
aca
ace
acl
acn
aco
acp
adg
adk
aeh
aeq
</code></pre>

<p><strong>Automatically reconstruct species trees by HCP</strong></p>
<pre><code class="bash">$ PhySpeTree autobuild -i organism_example_list.txt --hcp
Loading organisms names success.....

The result are store in:Outdata

Now loading data and constructing phylogenetic tree......
2016-10-29 19:44:11,660 KEGG INDEX DB INFO: Read organisms names success
2016-10-29 19:44:17,296 KEGG INDEX DB INFO: Retrieve and download of highly conserved protein 'Ribosomal protein L1' was successful store in p1.fasta file
2016-10-29 19:44:17,919 KEGG INDEX DB INFO: Retrieve and download of highly conserved protein 'DNA-directed RNA polymerase subunit alpha' was successful store in p2.fasta file
2016-10-29 19:44:18,369 KEGG INDEX DB INFO: Retrieve and download of highly conserved protein 'Leucyl-tRNA synthetase' was successful store in p3.fasta file
2016-10-29 19:44:18,943 KEGG INDEX DB INFO: Retrieve and download of highly conserved protein 'Metal-dependent proteases with chaperone activity' was successful store in p4.fasta file
2016-10-29 19:44:19,660 KEGG INDEX DB INFO: Retrieve and download of highly conserved protein 'Phenylalanine-tRNA synthethase alpha subunit' was successful store in p5.fasta file
2016-10-29 19:44:20,114 KEGG INDEX DB INFO: Retrieve and download of highly conserved protein 'Predicted GTPase probable translation factor' was successful store in p6.fasta file
2016-10-29 19:44:20,505 KEGG INDEX DB INFO: Retrieve and download of highly conserved protein 'Ribosomal protein L11' was successful store in p7.fasta file
2016-10-29 19:44:20,917 KEGG INDEX DB INFO: Retrieve and download of highly conserved protein 'Ribosomal protein L13' was successful store in p8.fasta file
2016-10-29 19:44:21,333 KEGG INDEX DB INFO: Retrieve and download of highly conserved protein 'Ribosomal protein L14' was successful store in p9.fasta file
......
</code></pre>

<p>Outputs:</p>
<pre><code>log.log
Outdata/
        RAxML_bestTree.T1
        RAxML_bipartitions.T1
        RAxML_bipartitionsBranchLabels.T1
        RAxML_bootstrap.T1
        RAxML_info.T1
   temp/
        conserved_protein20161029194411/
                                       　p1.fasta
                                       　p2.fasta
                                       　p3.fasta
                                       　......
        alignment20161029194429/
                                    p1.fasta
                                    p2.fasta
                                    p2.fasta
                                    ......
        concatenate20161029194432/
                                  concatenate.fasta
                                  concatenate.fasta-gb1
                                  concatenate.fasta-gb1.htm
                                  concatenate.fasta-gb1.phy
</code></pre>

<ul>
<li><code>log.log</code>: logs.</li>
<li>
<p><code>Outdata</code>: tree files. </p>
<ul>
<li><code>RAxML_bestTree.T1</code>: best ML search tree built by RAxML. </li>
<li><code>RAxML_bipartitions.T1</code>: bipartition tree built by RAxML.</li>
<li><code>RAxML_bipartitionsBranchLabels.T1</code>: bipartition tree by RAxML with branch length.</li>
<li><code>RAxML_bootstrap.T1</code>: bootstrap result.</li>
<li><code>RAxML_info.T1</code>: logs in running RAxML.</li>
</ul>
</li>
<li>
<p><code>temp</code>: temporary data used to check the quality of outputs in each step.</p>
<ul>
<li><code>conserved_protein</code>: highly conserved proteins retrieved from the KEGG database.</li>
<li><code>alignment</code>: aligned sequences.</li>
<li>
<p><code>concatenate</code>: concatenated sequences and conserved blocks.</p>
<ul>
<li><code>concatenate.fasta</code>: concatenated HCP sequences.</li>
<li><code>concatenate.fasta-gb1</code>: conserved blocks (by Gblocks).</li>
<li><code>concatenate.fasta-gb1.htm</code>: conserved blocks displayed in html.</li>
<li><code>concatenate.fasta-gb1.phy</code>: conserved blocks in the PHYLIP format.</li>
</ul>
</li>
</ul>
</li>
</ul>
<p><strong>Automatically reconstruct species trees by SSU rRNA</strong></p>
<pre><code class="bash">$ PhySpeTree autobuild -i organism_example_list.txt --srna
Loading organisms names success.....

The result are store in:Outdata

Now loading data and constructing phylogenetic tree......
2016-10-29 20:12:49,353 SSU rRNA DB INFO: Read organisms names success
2016-10-29 20:12:54,582 SSU rRNA DB INFO: Retrieve and download of organism 'aca' SSU rRNA sequence was successful
2016-10-29 20:12:56,831 SSU rRNA DB INFO: Retrieve and download of organism 'ace' SSU rRNA sequence was successful
2016-10-29 20:12:59,182 SSU rRNA DB INFO: Retrieve and download of organism 'acl' SSU rRNA sequence was successful
2016-10-29 20:13:01,545 SSU rRNA DB INFO: Retrieve and download of organism 'acn' SSU rRNA sequence was successful
2016-10-29 20:13:04,096 SSU rRNA DB INFO: Retrieve and download of organism 'aco' SSU rRNA sequence was successful
2016-10-29 20:13:06,972 SSU rRNA DB INFO: Retrieve and download of organism 'acp' SSU rRNA sequence was successful
2016-10-29 20:13:09,943 SSU rRNA DB INFO: Retrieve and download of organism 'adg' SSU rRNA sequence was successful
2016-10-29 20:13:12,707 SSU rRNA DB INFO: Retrieve and download of organism 'adk' SSU rRNA sequence was successful
2016-10-29 20:13:16,015 SSU rRNA DB INFO: Retrieve and download of organism 'aeh' SSU rRNA sequence was successful
2016-10-29 20:13:18,969 SSU rRNA DB INFO: Retrieve and download of organism 'aeq' SSU rRNA sequence was successful

</code></pre>

<p>Outputs:</p>
<pre><code>log.log
Outdata/
        RAxML_bestTree.T1
        RAxML_bipartitions.T1
        RAxML_bipartitionsBranchLabels.T1
        RAxML_bootstrap.T1
        RAxML_info.T1
   temp/
        rna_sequence20161029201249/
                                 rna_sequence.fasta

        rna_alignment20161029201319/
                                  rna_sequence.fasta
                                  rna_sequence.fasta-gb1
                                  rna_sequence.fasta-gb1.htm
                                  rna_sequence.fasta-gb1.phy
</code></pre>

<ul>
<li><code>log.log</code>: logs.</li>
<li><code>Outdata</code>: tree files like the HCP method.</li>
<li>
<p><code>temp</code>: temporary data used to check the quality of outputs in each step.</p>
<ul>
<li><code>rna_sequence</code>: SSU rRNA sequences retrieved from the SILVA database.</li>
<li>
<p><code>rna_alignment</code>: aligned sequences and conserved blocks.</p>
<ul>
<li>rna_sequence.fasta: aligned SSU rRNA sequences.</li>
<li>rna_sequence.fasta-gb1: conserved blocks (by Gblocks).</li>
<li>rna_sequence.fasta-gb1.htm: conserved blocks displayed in html.</li>
<li>rna_sequence.fasta-gb1.phy: conserved blocks in the PHYLIP format.</li>
</ul>
</li>
</ul>
</li>
</ul>
<h3 id="advanced-options">Advanced options</h3>
<p>Advanced options of internal software called in PhySpeTree can be set. These options are <code>enclosed in single quotes and start with a space</code>.</p>
<p>Here is an example of setting RAxML advanced options by <code>--raxml_p</code>:</p>
<pre><code class="bash">$ PhySpeTree autobuild -i organism_example_list.txt -o test --srna --raxml --raxml_p ' -f a -m GTRGAMMA  -p 12345 -x 12345 -# 100 -n T1'
</code></pre>

<p><strong>--muscle</strong></p>
<blockquote>
<p>Multiple sequence alignment by MUSCLE (default).</p>
</blockquote>
<p><strong>--muscle_p</strong></p>
<blockquote>
<p>Set MUSCLE advanced parameters, please see <a href="http://www.drive5.com/muscle/manual/options.html">MUSCLE Manual</a></p>
</blockquote>
<p>The default option:</p>
<table>
<thead>
<tr>
<th align="left">option</th>
<th align="left">description</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">-maxiter</td>
<td align="left">Maximum number of iterations to run. The default is 100.</td>
</tr>
</tbody>
</table>
<p><strong>--clustalw</strong></p>
<blockquote>
<p>Multiple sequence alignment by ClustalW2.</p>
</blockquote>
<p><strong>--clustalw_p</strong></p>
<blockquote>
<p>Set ClustalW2 advanced parameters, please see <a href="http://www.clustal.org/download/clustalw_help.txt">Clustalw Help</a>.</p>
</blockquote>
<p><strong>--mafft</strong></p>
<blockquote>
<p>Multiple sequence alignment by mafft.</p>
</blockquote>
<p><strong>--mafft_p</strong></p>
<blockquote>
<p>Set mafft advance parameters. Here use mafft default parameters, please see <a href="http://mafft.cbrc.jp/alignment/software/algorithms/algorithms.html">mafft algorithms</a></p>
</blockquote>
<p><strong>--gblocks</strong></p>
<blockquote>
<p>Trim by Gblocks.(default)</p>
</blockquote>
<p><strong>--gblocks_p</strong></p>
<blockquote>
<p>Set Gblocks advanced parameters, please see <a href="http://molevol.cmima.csic.es/castresana/Gblocks/Gblocks_documentation.html">Gblocks documentation</a>.</p>
</blockquote>
<p>The default option:</p>
<table>
<thead>
<tr>
<th align="left">option</th>
<th align="left">description</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">-t</td>
<td align="left">Choice type of sequence (default).</td>
</tr>
<tr>
<td align="left">-e</td>
<td align="left">Generic file extension. The default in PhySpeTree is "-gbl1".</td>
</tr>
</tbody>
</table>
<p><strong>--trimal</strong></p>
<blockquote>
<p>Trim by trimal.</p>
</blockquote>
<p><strong>--trimal_p</strong></p>
<blockquote>
<p>Set trimal advance parameters, please see<a href="http://trimal.cgenomics.org/use_of_the_command_line_trimal_v1.2">trimal command line</a></p>
</blockquote>
<p><strong>--ranxml</strong></p>
<blockquote>
<p>Reconstruct species tree by RAxML (default).</p>
</blockquote>
<p><strong>--raxml_p</strong></p>
<blockquote>
<p>Set RAxML advanced parameters, please see <a href="http://sco.h-its.org/exelixis/resource/download/NewManual.pdf">RAxML Manual</a>.</p>
</blockquote>
<p>The default option:</p>
<table>
<thead>
<tr>
<th align="left">option</th>
<th align="left">description</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">-f</td>
<td align="left">select algorithm. The default in PhySpeTree is <code>a</code>, rapid Bootstrap analysis and search for best­scoring ML tree in one program run.</td>
</tr>
<tr>
<td align="left">-m</td>
<td align="left">Model of binary (morphological), nucleotide, multi­State, or amino acid substitution. The PhySpeTree default set is PROTGAMMAJTTX.</td>
</tr>
<tr>
<td align="left">-p</td>
<td align="left">Specify a random number seed for the parsimony inferences. The default in PhySpeTree is 12345.</td>
</tr>
<tr>
<td align="left">-x</td>
<td align="left">Specify an integer number (random seed) and turn on rapid bootstrapping. The default in PhySpeTree is 12345.</td>
</tr>
<tr>
<td align="left">-N</td>
<td align="left">The same with -# specify the number of alternative runs on distinct starting trees. The default in PhySpeTree is 100.</td>
</tr>
</tbody>
</table>
<p><strong>--fasttree</strong></p>
<blockquote>
<p>Reconstruct species tree by FastTree.</p>
</blockquote>
<p><strong>--fasttree_p</strong></p>
<blockquote>
<p>Set FastTree advanced parameters, please see <a href="http://www.microbesonline.org/fasttree/">FastTree Helps</a>.</p>
</blockquote>
<p><strong>--iqtree</strong></p>
<blockquote>
<p>Reconstruct species tree by iqtree.</p>
</blockquote>
<p><strong>--iqtree_p</strong></p>
<blockquote>
<p>Set iqtree advanced parameters, please see <a href="http://www.microbesonline.org/fasttree/">IQ-TREE</a>.</p>
</blockquote>
<h2 id="build">build</h2>
<p>The <code>build</code> module is used to reconstruct species trees with manually prepared sequences. Advanced options are the same as <code>autobuild</code> module.</p>
<pre><code class="bash"># multiple method
$ PhySpeTree build -i example_hcp -o output --multiple

# single method
$ PhySpeTree build -i example_16s_ssurna.fasta -o output --single
</code></pre>

<h3 id="build-options">build options</h3>
<table>
<thead>
<tr>
<th align="left">option</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">-h</td>
<td align="left">Print help message and exits.</td>
</tr>
<tr>
<td align="left">-i</td>
<td align="left">Input a TXT file containing abbreviated species names.</td>
</tr>
<tr>
<td align="left">-o</td>
<td align="left">A directory to store outputs. The default is "Outdata".</td>
</tr>
<tr>
<td align="left">-t</td>
<td align="left">Number of processing threads (CPUs). The default is 1.</td>
</tr>
<tr>
<td align="left">--multiple</td>
<td align="left">Specify concatenate highly conserved protein method to reconstruct phylogenetic tree.</td>
</tr>
<tr>
<td align="left">--single</td>
<td align="left">Use SSU rRNA data to reconstruct phylogenetic tree.</td>
</tr>
</tbody>
</table>
<h3 id="example_1">Example</h3>
<p><strong>Build species trees by manually prepared HCP</strong></p>
<p>The HCP sequences belonging to the same class are prepared in one FASTA format file, and all FASTA format files are stored in the same folder. For example, the folder <a href="../example/example_build_hcp.tar.gz">example_build_hcp</a> contains 10 classes of HCP (p1~p10) corresponding to 10 different organisms. There is no limit number of HCP sequences. We recommend no less than 10 highly conserved proteins to ensure the accuracy of the reconstructed phylogenetic tree.</p>
<p>Download and decompress the example input file:</p>
<pre><code class="bash">
$ wget &quot;https://yangfangs.github.io/physpetools/example/example_build_hcp.tar.gz&quot;

--2016-10-29 20:40:41--  https://yangfangs.github.io/physpetools/example/example_build_hcp.tar.gz
Resolving yangfangs.github.io (yangfangs.github.io)... 151.101.48.133
Connecting to yangfangs.github.io (yangfangs.github.io)|151.101.48.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 17419 (17K) [application/octet-stream]
Saving to: ‘example_build_hcp.tar.gz’

example_build_hcp.tar.gz     100%[==============================================&gt;]  17.01K  --.-KB/s    in 0.009s  

2016-10-29 20:40:42 (1.92 MB/s) - ‘example_build_hcp.tar.gz’ saved [17419/17419]


$ tar -zxvf example_build_hcp.tar.gz

example_build_hcp/                                                                                                   
example_build_hcp/p1.fasta
example_build_hcp/p2.fasta
example_build_hcp/p3.fasta
example_build_hcp/p4.fasta
example_build_hcp/p5.fasta
example_build_hcp/p6.fasta
example_build_hcp/p7.fasta
example_build_hcp/p8.fasta
example_build_hcp/p9.fasta
example_build_hcp/p10.fasta
</code></pre>

<p>Check HCP:</p>
<pre><code class="bash">$ cd  example_build_hcp/
$ cat p1.fasta 
&gt;aeh
MARLTKRQKAIREKIDPAQQYPVAEALGLLRELPGAKFTESVEVAVNLGVDPRKSDQIVR
GSTVLPNGTGKTVRVAVFAQGDAAEAAKEAGADIVGMDDLAEQVKGGNLDFDVVVAAPDA
MGVVGRLGPILGPRGLMPNPKVGTVTPDVAGAVKNAKAGQVRYRTDKGGIIHCAIGKVDF
EVEALQQNLQALITDLQKLKPANSKGVYLKKVAVSTTMGPGLAVDLASLET
&gt;adk
MAKLTKKQKAQQGKVDSTKLYPFAEAVALVKEAATAKFDESIDVAVQLGVDAKKSDQVVR
GAVVLPNGTGKTTRVAVFAQGAKAEEAKAAGADVVGMDDLAAQVKAGDMPFDVVIAAPDA
MRVVGTLGQILGPRGLMPNPKVGTVTPDVATAVKNAKAGQVQFRVDKAGIVHTTIGRRSF
ADDKLQGNLAALIEALNKAKPATSKGVYLRKVAVSSTMGVGVRVDTQSIAA
&gt;acp
MAHVAKKYKAAAEKVDRTKRYKLDEAMSLVKQTATKKFDETVDASINLGVDPKHADQVVR
GAVVLPHGMGKTVRLAVFAKGDKAKEAQEAGADIVGAEDLAEKIQGGFMDFDKLIATPDM
MGVVGRLGKILGPRGLMPNPKVGTVTMDLARAVKEQKAGKVEFRVEKAGIVHVPFGKASF
DPDKLKANFSAIMEVIYKAKPQTAKGVYVKNVTLSTTMGPGIKVDLAELAAQHA
&gt;acn
MSGDGSSYSAEEGIRELLQSAKAKFRESVDVAIKLSVADSKSGESVRGAVVLPKGLGREV
RVAVFAKGEHAKHASDAGADVVGDEELIEEIKKGRKLDVDWCIATPDFMPQISAIAKILG
PRGLMPNPKFGTVTLELAKMVGVIKSGQVKFKSDRYGIVHVKIGDVSFTPEDLLENFNAV
VVAVQNLKPATIKGSYVRGVFVNSTMGRSFRIAGIG
&gt;adg
MPKHGKKYLEAKKQVDRTKLYDPYEALELVKRLASAKFDETVEVAVRLGVDPRHADQQVR
GAVVLPHGTGKTRRVLVFARGEKAKEAEAAGADYVGAEDLIARIQGGWLDFDVAIATPDM
MAMVGRIGRILGPRGLMPNPKTGTVTFDVAQAVAEAKAGRVEYRTDKAGIVHAPIGKVSF
EVEKLVENLKALVDALVRAKPPAAKGQYLRSITVSSTMGPGVKVNPAKLLAS
&gt;acl
MKRGKKYLEAVKLYDKSVAYTGLEAVELAKKTSVAKFDATVEVAFRLNVDPRKADQNLRG
AISLPHGTGKTVRVVVIAKPEKAKEALAAGALEAGDVELIDKIGKGWFDFDVMVATPDMM
AQLGKLGRVLGPKGLMPNPKTGTVTLDVAKAVEEIKAGKIEYRTDKVGNIHAPIGKVSFD
SNKLHENMLAIYNQLVRIKPATVKGTYIKKIALSTTMGPGIMVEENNIKK
&gt;ace
MKRGKKYRAAAQLVDRTKLYSPLEAMRLAKQTNTMRVPATVEVAMRLGVDPRKADQMVRG
TVNLPHGTGKTPRVLVFATAERAEEARAAGADYVGADELIEQVANGFLDFDAVVATPDLM
GKVGRLGRILGPRGLMPNPKTGTVTNDVAKAVADIKSGKIEFRVDRQANLHLVIGKTDFT
EQQLVENYAAALDEVLRLKPPTAKGRYLKKVTISTTMGPGIPVDPNRVRNLLAEETAAA
&gt;aeq
MTKHGKKYVEAEKQIPAEPVSPLAAMKLLKEISVANFDETVTGDFRLGIDTRQADQQLRG
TVSLPNGSGKTVRVAVFAEGAAAQAAEEAGADIVGTDELMQQIQAGEFNFDAAVATPDQM
GKVGRLGKILGPRGLMPNPKLGTVTNDVAKAIKELKGGRVEYRADRYGIAHVVLGKVSFT
PEQLAENYGAVYDEILRMKPAAAKGKYVKSITVSGTMTPGVSVDSSVTRAYTESAE
&gt;aca
MSKKVSKNVAKARAAVEPRPYTLQDAVPLLQQVKFAKFDETVDLTMRLGVDPRHADQMVR
GTVVLPHGLGKTKKVAVITTGDRQKEAEAAGAEIVGGEELVEKIQKESWTDFDALIATPD
MMRSVGRLGKVLGPRGLMPNPKTGTVTNDVAAAVKEIKAGKIEYRTDKTALVHVPVGKLS
FPAEKLIDNAMTVITSVVRAKPSAAKGKYIKGITLSSTMGPGIPLDGSVADAAAKA
&gt;aco
MAKKSKRYSEIAAKVDSTKLYGLREAVDLYKEVATAKFDESLEVHIRLGVDPRHADQQVR
GTIVLPHGTGITKRVLVLAVGEKVKEAEDAGADIVGGDDLIQKISTGWLDFDAVIATPDM
MKSVGRLGKILGPRGLMPSAKAGTVTFDVADAIKEIKAGRVEFRVDKTAIIHNMVGKKSF
EAEKLFENLKVLYRAILKARPASAKGTYVRSFYIAPTMGVGIKIDPVAASKEVAEA
</code></pre>

<p>Reconstruct species tree and store outputs in the <code>build_hcp_tree</code> folder:</p>
<pre><code class="bash">PhySpeTree build -i example_build_hcp -o build_hcp_tree --multiple
</code></pre>

<p><strong>Build species trees by manually prepared SSU rRNA</strong></p>
<p>All SSU rRNA sequences are prepared in one FASTA format file, for example <a href="../example/example_build_srna.fasta">example_build_srna</a>.</p>
<p>Download and decompress the example input file:</p>
<pre><code>$ wget &quot;https://yangfangs.github.io/physpetools/example/example_build_srna.fasta&quot;

--2016-10-29 20:56:31--  https://yangfangs.github.io/physpetools/example/example_build_srna.fasta
Resolving yangfangs.github.io (yangfangs.github.io)... 151.101.48.133
Connecting to yangfangs.github.io (yangfangs.github.io)|151.101.48.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 14982 (15K) [application/octet-stream]
Saving to: ‘example_build_srna.fasta’

example_build_srna.fasta     100%[==============================================&gt;]  14.63K  --.-KB/s    in 0.005s  

2016-10-29 20:56:33 (3.14 MB/s) - ‘example_build_srna.fasta’ saved [14982/14982]

$ grep '&gt;' example_build_srna.fasta 
&gt;aca
&gt;ace
&gt;acl
&gt;acn
&gt;aco
&gt;acp
&gt;adg
&gt;adk
&gt;aeh
&gt;aeq
</code></pre>

<p>Reconstruct species tree and store outputs in the <code>build_srna_tree</code> folder:</p>
<pre><code class="bash">PhySpeTree build -i example_build_srna.fasta -o build_srna_tree --single --fasttree
</code></pre>

<h2 id="combine">combine</h2>
<p>The <strong>combine</strong> module is used to combine trees generated from different methods. It contains two steps, at first merge different tree files into the same file. You can use <code>cat</code> bash command in the Linux system, for example:</p>
<pre><code class="bash">$ cat tree1.tree tree2.tree &gt; combineTree.tree
</code></pre>

<p>Then, use <strong>combine</strong>:</p>
<pre><code class="bash">$ PhySpeTree combine -i combineTree.tree [options]*
</code></pre>

<h3 id="combine-options">combine options</h3>
<table>
<thead>
<tr>
<th>option</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>-h</td>
<td>Print help message and exits.</td>
</tr>
<tr>
<td>-i</td>
<td>Input PHYLIP format file containing multiple trees.</td>
</tr>
<tr>
<td>-o</td>
<td>Output directory. The default is "combineTree".</td>
</tr>
<tr>
<td>--mr</td>
<td>Majority rule trees.</td>
</tr>
<tr>
<td>--mre</td>
<td>Extended majority rule trees.</td>
</tr>
<tr>
<td>--strict</td>
<td>Strict consensus trees.</td>
</tr>
<tr>
<td>--astral</td>
<td>Use ASTRAL combine multi gene tree.</td>
</tr>
</tbody>
</table>
<h3 id="example_2">Example</h3>
<p><a href="../example/example_combine_tree.tar.gz">example_combine_tree.tar.gz</a> contains <code>tree1.tree</code> and <code>tree2.tree</code> reconstructed by the HCP and SSU rRNA method, respectively.</p>
<p>Download and decompress the example input file:</p>
<pre><code class="bash">
$ wget &quot;https://yangfangs.github.io/physpetools/example/example_combine_tree.tar.gz&quot;

--2016-10-30 13:32:06--  https://yangfangs.github.io/physpetools/example/example_combine_tree.tar.gz
Resolving yangfangs.github.io (yangfangs.github.io)... 151.101.48.133
Connecting to yangfangs.github.io (yangfangs.github.io)|151.101.48.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 661 [application/octet-stream]
Saving to: ‘example_combine_tree.tar.gz’

example_combine_tree.tar.gz  100%[==============================================&gt;]     661  --.-KB/s    in 0s      

2016-10-30 13:32:07 (380 MB/s) - ‘example_combine_tree.tar.gz’ saved [661/661] 


$ tar -zxvf example_combine_tree.tar.gz 
example_combine_tree/
example_combine_tree/tree2.tree
example_combine_tree/tree1.tree
</code></pre>

<p>Merge <code>tree1.tree</code> and <code>tree2.tree</code>:</p>
<pre><code class="bash">$ cd example_combine_tree/
$ cat tree1.tree tree2.tree &gt; combine.tree
</code></pre>

<p>Combine trees:</p>
<pre><code class="bash">PhySpeTree combine -i combine.tree -o combineTree
</code></pre>

<p>Outputs:</p>
<pre><code>combine/
        RAxML_info.T1
        RAxML_MajorityRuleConsensusTree.T1
</code></pre>

<ul>
<li><code>RAxML_info.T1</code>: logs in running RAxML.</li>
<li><code>RAxML_MajorityRuleConsensusTree.T1</code>: the majority rule consensus tree.</li>
</ul>
<h3 id="using-astral-option">Using <code>--astral</code> option</h3>
<blockquote>
<p>Notice: The --astral option calls the third-party software ASTRAL. Please be aware that JRE has been installed in your running environments.
For users who run the Docker image of PhySpeTree, JRE is unnecessary.</p>
</blockquote>
<pre><code class="bash">PhySpeTree combine -i combine.tree -o combineTree --astral
</code></pre>

<p>Outputs:</p>
<pre><code>combineTree/
           combine.tree
</code></pre>

<h2 id="iview">iview</h2>
<p>PhySpeTree provides the <code>iview</code> module to annotate taxonomic information (kingdom, phylum, class, or order) of output trees and to generate configure files linked to <a href="http://itol.embl.de/">iTol</a>.</p>
<pre><code class="bash">$ PhySpeTree iview -i organism_example_list.txt --range
</code></pre>

<h3 id="iview-options">iview options</h3>
<table>
<thead>
<tr>
<th align="left">option</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">-h</td>
<td align="left">Print help message and exits.</td>
</tr>
<tr>
<td align="left">-i</td>
<td align="left">Input a TXT file containing abbreviated species names.</td>
</tr>
<tr>
<td align="left">-o</td>
<td align="left">A directory to store outputs. The default is "iview".</td>
</tr>
<tr>
<td align="left">-a</td>
<td align="left">Colored ranges [kingdom, phylum, class or order].</td>
</tr>
<tr>
<td align="left">-r/--range</td>
<td align="left">Annotating labels with ranges by kingdom, phylum, class or order. The default is phylum.</td>
</tr>
<tr>
<td align="left">-c/--color</td>
<td align="left">Annotating labels without ranges by kingdom, phylum, class or order. The default is phylum.</td>
</tr>
<tr>
<td align="left">-l/--labels</td>
<td align="left">Change species labels from abbreviated names to full names.</td>
</tr>
</tbody>
</table>
<h3 id="example_3">Example</h3>
<p>Download the example file:</p>
<pre><code class="bash">$ wget &quot;https://yangfangs.github.io/physpetools/example/organism_example_list.txt&quot;

--2016-10-30 13:40:48--  https://yangfangs.github.io/physpetools/example/organism_example_list.txt
Resolving yangfangs.github.io (yangfangs.github.io)... 151.101.48.133
Connecting to yangfangs.github.io (yangfangs.github.io)|151.101.48.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 39 [text/plain]
Saving to: ‘organism_example_list.txt’

organism_example_list.txt    100%[==============================================&gt;]      39  --.-KB/s    in 0s      

2016-10-30 13:40:50 (21.5 MB/s) - ‘organism_example_list.txt’ saved [39/39]
</code></pre>

<p><strong>Annotate the tree by kingdom</strong></p>
<pre><code class="bash">$ PhySpeTree iview -i organism_example_list.txt --range -a kingdom
Color range by kingdom was complete.
</code></pre>

<p>The color range file is store in the <code>iview</code> folder:</p>
<pre><code class="bash">$ cd iview
$ cat range_color_by_kingdom.txt 
TREE_COLORS
SEPARATOR TAB
DATA
aca     range   #BEBF5A Prokaryotes
ace     range   #BEBF5A Prokaryotes
acl     range   #BEBF5A Prokaryotes
acn     range   #BEBF5A Prokaryotes
aco     range   #BEBF5A Prokaryotes
acp     range   #BEBF5A Prokaryotes
adg     range   #BEBF5A Prokaryotes
adk     range   #BEBF5A Prokaryotes
aeh     range   #BEBF5A Prokaryotes
aeq     range   #BEBF5A Prokaryotes
</code></pre>

<p><strong>Annotate the tree by phylum</strong></p>
<pre><code class="bash">$ PhySpeTree iview -i organism_example_list.txt --range -a phylum
Color range by phylum was complete.
</code></pre>

<p>The color range file is store in the <code>iview</code> folder:</p>
<pre><code class="bash">$ cd iview
$ cat range_color_by_phylum.txt 
TREE_COLORS
SEPARATOR TAB
DATA
aca     range   #865142 Bacteria
ace     range   #865142 Bacteria
acl     range   #865142 Bacteria
acn     range   #865142 Bacteria
aco     range   #865142 Bacteria
acp     range   #865142 Bacteria
adg     range   #865142 Bacteria
adk     range   #865142 Bacteria
aeh     range   #865142 Bacteria
aeq     range   #865142 Bacteria
</code></pre>

<p><strong>Annotate the tree by class</strong></p>
<pre><code class="bash">$ PhySpeTree iview -i organism_example_list.txt --range -a class
Color range by class was complete.
</code></pre>

<p>The color range file is store in the <code>iview</code> folder:</p>
<pre><code class="bash">$ cd iview
$ cat range_color_by_class.txt 
TREE_COLORS
SEPARATOR TAB
DATA
aca     range   #9AB7F3 Acidobacteria
ace     range   #99D1DB Actinobacteria
acl     range   #A5E58D Tenericutes
acn     range   #94F1C1 Alphaproteobacteria
aco     range   #D67A21 Synergistetes
acp     range   #DD9284 Deltaproteobacteria
adg     range   #3E70B8 Firmicutes - Clostridia
adk     range   #DDC8B7 Betaproteobacteria
aeh     range   #72E137 Gammaproteobacteria - Others
aeq     range   #99D1DB Actinobacteria
</code></pre>

<p><strong>Annotate the tree by order</strong></p>
<pre><code class="bash">$ PhySpeTree iview -i organism_example_list.txt --range -a order
Color range by order was complete.
</code></pre>

<p>The color range file is store in the <code>iview</code> folder:</p>
<pre><code class="bash">$ cd iview
$ cat range_color_by_order.txt 
TREE_COLORS
SEPARATOR TAB
DATA
aca     range   #AA8761 Acidobacterium
ace     range   #8770BC Acidothermus
acl     range   #3BD26B Acholeplasma
acn     range   #D1B487 Anaplasma
aco     range   #D96D21 Aminobacterium
acp     range   #AC4E16 Anaeromyxobacter
adg     range   #287AD8 Ammonifex
adk     range   #C8184E Alicycliphilus
aeh     range   #57A569 Alkalilimnicola
aeq     range   #F1A2B7 Adlercreutzia
</code></pre>

<h2 id="check">check</h2>
<p>The <code>check</code> module is used to check whether input organisms are in pre-built databases.</p>
<pre><code class="bash">$ PhySpeTree check -i organism_example_list.txt --ehcp
</code></pre>

<h3 id="check-options">check options</h3>
<table>
<thead>
<tr>
<th align="left">option</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">-h</td>
<td align="left">Print help message and exits.</td>
</tr>
<tr>
<td align="left">-i</td>
<td align="left">Input a TXT file containing abbreviated species names.</td>
</tr>
<tr>
<td align="left">-o</td>
<td align="left">A directory to store outputs. The default is "check".</td>
</tr>
<tr>
<td align="left">--hcp</td>
<td align="left">Check whether organisms are supported in the KEGG database.</td>
</tr>
<tr>
<td align="left">--ehcp</td>
<td align="left">Check input organisms prepare for extend autobuild tree module.</td>
</tr>
<tr>
<td align="left">--srna</td>
<td align="left">Check whether organisms are supported in the SILVA database.</td>
</tr>
</tbody>
</table>
<h3 id="example_4">Example</h3>
<p><strong>Check extended organisms in <code>autobuild</code></strong></p>
<p>Determine proteins to be prepared in the <code>autobuild</code> module with the <code>--ehcp</code> option, for example, <a href="../example/organism_example_list.txt">organism_example_list.txt</a></p>
<p>Download the example file:</p>
<pre><code class="bash">$ wget &quot;https://yangfangs.github.io/physpetools/example/organism_example_list.txt&quot;

--2016-10-30 13:40:48--  https://yangfangs.github.io/physpetools/example/organism_example_list.txt
Resolving yangfangs.github.io (yangfangs.github.io)... 151.101.48.133
Connecting to yangfangs.github.io (yangfangs.github.io)|151.101.48.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 39 [text/plain]
Saving to: ‘organism_example_list.txt’

organism_example_list.txt    100%[==============================================&gt;]      39  --.-KB/s    in 0s      

2016-10-30 13:40:50 (21.5 MB/s) - ‘organism_example_list.txt’ saved [39/39]
</code></pre>

<p>Check: </p>
<pre><code class="bash">$ PhySpeTree check -i organism_example_list.txt --ehcp

'Ribosomal protein L1' ----------------------------------&gt; p1.fasta

'DNA-directed RNA polymerase subunit alpha' ----------------------------------&gt; p2.fasta

'Leucyl-tRNA synthetase' ----------------------------------&gt; p3.fasta

'Metal-dependent proteases with chaperone activity' ----------------------------------&gt; p4.fasta

'Phenylalanine-tRNA synthethase alpha subunit' ----------------------------------&gt; p5.fasta

'Predicted GTPase probable translation factor' ----------------------------------&gt; p6.fasta

'Ribosomal protein L11' ----------------------------------&gt; p7.fasta

'Ribosomal protein L13' ----------------------------------&gt; p8.fasta

'Ribosomal protein L14' ----------------------------------&gt; p9.fasta

'Ribosomal protein L22' ----------------------------------&gt; p10.fasta

'Ribosomal protein L3' ----------------------------------&gt; p11.fasta

'Ribosomal protein L5' ----------------------------------&gt; p12.fasta

'Ribosomal protein S11' ----------------------------------&gt; p13.fasta

'Ribosomal protein S17' ----------------------------------&gt; p14.fasta

'Ribosomal protein S2' ----------------------------------&gt; p15.fasta

'Ribosomal protein S3' ----------------------------------&gt; p16.fasta

'Ribosomal protein S4' ----------------------------------&gt; p17.fasta

'Ribosomal protein S5' ----------------------------------&gt; p18.fasta

'Ribosomal protein S7' ----------------------------------&gt; p19.fasta

'Ribosomal protein S8' ----------------------------------&gt; p20.fasta

'Ribosomal protein S9' ----------------------------------&gt; p21.fasta

'Seryl-tRNA synthetase' ----------------------------------&gt; p22.fasta

'Arginyl-tRNA synthetase' ----------------------------------&gt; p23.fasta

'DNA-directed RNA polymerase beta subunit' ----------------------------------&gt; p24.fasta

'Ribosomal protein S13' ----------------------------------&gt; p25.fasta

Check extend highly conserved protein is completed.

</code></pre>

<p>The check result is stored in the <code>check</code> folder. In <code>physpe_echp_extend.txt</code> file indicates class of HCP and their corresponding names, which will be used to prepare extended HCP sequences.</p>
<pre><code class="bash">$ cd check
$ cat physpe_echp_extend.txt 
'Ribosomal protein L1' ----------------------------------&gt; p1.fasta
'DNA-directed RNA polymerase subunit alpha' ----------------------------------&gt; p2.fasta
'Leucyl-tRNA synthetase' ----------------------------------&gt; p3.fasta
'Metal-dependent proteases with chaperone activity' ----------------------------------&gt; p4.fasta
'Phenylalanine-tRNA synthethase alpha subunit' ----------------------------------&gt; p5.fasta
'Predicted GTPase probable translation factor' ----------------------------------&gt; p6.fasta
'Ribosomal protein L11' ----------------------------------&gt; p7.fasta
'Ribosomal protein L13' ----------------------------------&gt; p8.fasta
'Ribosomal protein L14' ----------------------------------&gt; p9.fasta
'Ribosomal protein L22' ----------------------------------&gt; p10.fasta
'Ribosomal protein L3' ----------------------------------&gt; p11.fasta
'Ribosomal protein L5' ----------------------------------&gt; p12.fasta
'Ribosomal protein S11' ----------------------------------&gt; p13.fasta
'Ribosomal protein S17' ----------------------------------&gt; p14.fasta
'Ribosomal protein S2' ----------------------------------&gt; p15.fasta
'Ribosomal protein S3' ----------------------------------&gt; p16.fasta
'Ribosomal protein S4' ----------------------------------&gt; p17.fasta
'Ribosomal protein S5' ----------------------------------&gt; p18.fasta
'Ribosomal protein S7' ----------------------------------&gt; p19.fasta
'Ribosomal protein S8' ----------------------------------&gt; p20.fasta
'Ribosomal protein S9' ----------------------------------&gt; p21.fasta
'Seryl-tRNA synthetase' ----------------------------------&gt; p22.fasta
'Arginyl-tRNA synthetase' ----------------------------------&gt; p23.fasta
'DNA-directed RNA polymerase beta subunit' ----------------------------------&gt; p24.fasta
'Ribosomal protein S13' ----------------------------------&gt; p25.fasta
</code></pre>

<p><strong>Check whether input organisms are supported in PhySpeTree</strong></p>
<p>Check whether input species are supported by the KEGG database when using the <code>--hcp</code> method, for example <a href="../example/191speciesnames.txt">example download</a>.</p>
<p>Download the example file:</p>
<pre><code class="bash">$ wget &quot;https://yangfangs.github.io/physpetools/example/191speciesnames.txt&quot;

--2016-10-30 14:48:21--  https://yangfangs.github.io/physpetools/example/191speciesnames.txt
Resolving yangfangs.github.io (yangfangs.github.io)... 151.101.48.133
Connecting to yangfangs.github.io (yangfangs.github.io)|151.101.48.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 773 [text/plain]
Saving to: ‘191speciesnames.txt’

191speciesnames.txt          100%[==============================================&gt;]     773  --.-KB/s    in 0s      

2016-10-30 14:48:22 (322 MB/s) - ‘191speciesnames.txt’ saved [773/773]
</code></pre>

<p>The check results show one organism named 'ges' is not supported in PhySpeTree:</p>
<pre><code class="bash">$ PhySpeTree check -i 191speciesnames.txt --hcp
WARNING: The following species are not supported by KEGG DATABASE:
ges
Checked  whether the input species names in KEGG DATABASE completed.
</code></pre>

<p>Check whether input species are supported by SILVA database when using the <code>--srna</code> metho, for example <a href="../example/191speciesnames.txt">example download</a></p>
<p>Download the example file:</p>
<pre><code class="bash">$ wget &quot;https://yangfangs.github.io/physpetools/example/191speciesnames.txt&quot;

--2016-10-30 14:48:21--  https://yangfangs.github.io/physpetools/example/191speciesnames.txt
Resolving yangfangs.github.io (yangfangs.github.io)... 151.101.48.133
Connecting to yangfangs.github.io (yangfangs.github.io)|151.101.48.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 773 [text/plain]
Saving to: ‘191speciesnames.txt’

191speciesnames.txt          100%[==============================================&gt;]     773  --.-KB/s    in 0s      

2016-10-30 14:48:22 (322 MB/s) - ‘191speciesnames.txt’ saved [773/773]
</code></pre>

<p>The check results show 28 organisms are not supported in PhySpeTree:</p>
<pre><code class="bash">
(progect) [yangfang@localhost test_check] $ PhySpeTree check -i 191speciesnames.txt --srna
WARNING: The following species are not supported by SILVA DATABASE:
neq
ape
tac
mmp
gla
tps
cho
ddi
spo
aga
tru
mpu
lin
ban
bce
ljo
san
spg
ges
lis
sco
cdi
mle
wsu
rpr
bpe
bpa
ppr
Checked  whether the input species names in SILVA DATABASE completed.
</code></pre>

<p>For organisms not in the pre-built list, PhySpeTree provides extend options (<code>--echp</code> or <code>--esrna</code>) to insert manually prepared sequences.</p></div>
        </div>

        <footer class="col-md-12">
            <hr>
            <p>Documentation built with <a href="https://www.mkdocs.org/">MkDocs</a>.</p>
        </footer>
        <script>
            var base_url = "..",
                shortcuts = {"help": 191, "next": 78, "previous": 80, "search": 83};
        </script>
        <script src="../js/base.js" defer></script>
        <script src="../search/main.js" defer></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
    <div class="modal-dialog">
        <div class="modal-content">
            <div class="modal-header">
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                <h4 class="modal-title" id="exampleModalLabel">Search</h4>
            </div>
            <div class="modal-body">
                <p>
                    From here you can search these documents. Enter
                    your search terms below.
                </p>
                <form role="form">
                    <div class="form-group">
                        <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query" title="Type search term here">
                    </div>
                </form>
                <div id="mkdocs-search-results"></div>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div><div class="modal" id="mkdocs_keyboard_modal" tabindex="-1" role="dialog" aria-labelledby="Keyboard Shortcuts Modal" aria-hidden="true">
    <div class="modal-dialog">
        <div class="modal-content">
            <div class="modal-header">
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                <h4 class="modal-title" id="exampleModalLabel">Keyboard Shortcuts</h4>
            </div>
            <div class="modal-body">
              <table class="table">
                <thead>
                  <tr>
                    <th style="width: 20%;">Keys</th>
                    <th>Action</th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td class="help shortcut"><kbd>?</kbd></td>
                    <td>Open this help</td>
                  </tr>
                  <tr>
                    <td class="next shortcut"><kbd>n</kbd></td>
                    <td>Next page</td>
                  </tr>
                  <tr>
                    <td class="prev shortcut"><kbd>p</kbd></td>
                    <td>Previous page</td>
                  </tr>
                  <tr>
                    <td class="search shortcut"><kbd>s</kbd></td>
                    <td>Search</td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div>

    </body>
</html>
